Makefile

```diff
@@ -132,12 +132,6 @@ docker-push:
 # Rules for verification, formatting, linting, testing and cleaning #
 #####################################################################
 
-.PHONY: revendor
-revendor:
-	@GO111MODULE=on go mod tidy
```

Comment:
We might need to add a new target for just `go mod tidy`.

Reply:
Isn't there a check that the PR is "clean" after all the generations are run? But yeah it might make sense to have something like that + and updated dev guide.

Reply:
I added a `make tidy` to make sure that the code especially the `go.sum` file is clean. I also updated the `dependencies.md` to reflect those changes.

Reply:
Yeah, the check you removed in `hack/check-generate.sh` was exactly this. We can reintroduce it to run the new target that I proposed above.